### DISCUSSION TITLE
Spreading awareness of PEP 752

### DISCUSSION THREAD

**Discussion by ofek** [apache/airflow#41657] [Ideas]
Hello team! I just wanted to bring to your attention a new PEP about package repository namespaces where I used your packages/community as a motivating example 🙂

We would love your feedback in the current discussion: https://discuss.python.org/t/pep-752-package-repository-namespaces/61227

cc @potiuk

**Comment by potiuk** [comment]
This is cool. Actually we had a very relevant discussion today about a potential confusion and new provider created by someone - see https://github.com/apache/airflow/pull/41556#issuecomment-2302037816  - I will ask other PMC members to take a look. After quick reaing this would be great to be able to have it for Airflow.

**Comment by ashb** [comment]
Very nice. In some ways I almost don't want us to reserve that prefix so that other members of the community can release their own packages.

Maybe that's okay, and we say to those people "use `airflow-provider-` prefix, and reserve `apache-airflow-` for official ones"

**Reply by potiuk** [reply]
> Maybe that's okay, and we say to those people "use airflow-provider- prefix, and reserve apache-airflow- for official ones" 

Yes. That would be nice if we could make it a convention and have a way to enforce the "root" namespace to only be used by Apache Airflow team.  This is also the "expectation" of the ASF - naming convention of packages is part of "branding" expectations from project - and currently with PyPI is not enforceable - with that change however it could be.

**Reply by ofek** [reply]
Hey! Two points:

1. You are not a corporate organization so the namespace would be public meaning your packages will get a special icon on PyPI but others can still release, and no icon.
2. If I change the spec to allow community projects to have private namespaces what you're saying is possible because child grants can be public even if the root is private.

**Reply by ofek** [reply]
Oh wait, I guess Apache may pay for this because it's a large entity? If not then I would have to change a line in the PEP to allow non-paying organizations to have private namespaces.

**Reply by potiuk** [reply]
I think ASF **might not** be able to pay for it, but maybe the "Corporate" status of the ASF might be donated by the PSF - we have similar thing with Slack, GitHub and others. 

While ASF is a big organization, with > 200 projects, and ~ 1000 members and many thousands committers,  it's purely volunteer-driven with practically zero friction to get new contributors, maintainer sand we do not have deep pockets, so any kind (especially "per-user") payment is generally no-go.

**Reply by ofek** [reply]
That's a great idea actually, I could add an explicit exception that reviewers could grant community projects a private namespace at their discretion.

**Comment by ofek** [comment]
The policy/operational parts have been extracted into a separate PEP:

- Specification: [PEP 752](https://peps.python.org/pep-0752/)
- Policy: [PEP 755](https://peps.python.org/pep-0755/)

Would either of you care to comment on the discussions? We are actively trying to assess community reception to the proposal and it is difficult without explicit feedback!

**Comment by potiuk** [comment]
Sorry I was on vacations - and only got to it now @ofek  - but I commented now https://discuss.python.org/t/pep-755-implicit-namespace-policy-for-pypi/63191/75  . I hope adding an explanation from the perspective an open-source organisation/PMC that would need this feature and why might help as a 'potential user' case.
